src/distdl/nn/conv_feature.py

```diff
@@ -249,10 +263,10 @@ def _distdl_module_setup(self, input):
 
         # Using that information, we can get there rest of the halo information
         exchange_info = self._compute_exchange_info(x_global_shape_after_pad,
-                                                    self.kernel_size,
-                                                    self.stride,
-                                                    0,
-                                                    self.dilation,
+                                                    self._expand_parameter(self.kernel_size),
```

Comment:
Each of these should really be done when, e.g., `self.kernel_size` is set (`self.kernel_size = self._expand_parameter(kernel_size)`.  You can see that the tuple-ified data is what torch stores with an integer is specified.

Reply:
I could get rid of `_expand_parameter` as it only needs to be used for the padding parameter just once. I was feeling a bit wary of relying on PyTorch's internal variables though, as it would make debugging a breaking change to the PyTorch api a bit more challenging. I don't think they explicitly document anything about this internal behavior.

Reply:
They do not, but what I'm proposing is that you follow this pattern:

```
self.kernel_size = _expand_parameters(kernel_size)
...
layer = TorchConvLayer(..., kernel_size=self.kernel_size, ...)

```